File: web/universal-binding/lib/mods/euler.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var Euler = function () {
  function Euler(x, y, z, order) {
    _classCallCheck(this, Euler);

    this.isEuler = true;
    this._x = 0;
    this._y = 0;
    this._z = 0;

    this._x = x || 0;
    this._y = y || 0;
    this._z = z || 0;
    this._order = order || Euler.DefaultOrder;
  }

  _createClass(Euler, [{
    key: 'set',
    value: function set(x, y, z, order) {
      this._x = x;
      this._y = y;
      this._z = z;
      this._order = order || this._order;
      this.onChangeCallback();
      return this;
    }
  }, {
    key: 'onChange',
    value: function onChange(callback) {
      this.onChangeCallback = callback;
      return this;
    }
  }, {
    key: 'onChangeCallback',
    value: function onChangeCallback() {}
  }, {
    key: 'order',
    get: function get() {
      return this._order;
    },
    set: function set(value) {
      this._order = value;
      this.onChangeCallback();
    }
  }]);

  return Euler;
}();

Euler.RotationOrders = ['XYZ', 'YZX', 'ZXY', 'XZY', 'YXZ', 'ZYX'];
Euler.DefaultOrder = 'XYZ';
exports.default = Euler;
module.exports = exports['default'];

File: iOS/sdk/Gypo/Euler.m
//
//  Euler.m
//  AliWeex
//
//  Created by 对象 on 2017/9/19.
//

#import "Euler.h"

#define ROTATION_ORDERS (@[@"XYZ",@"YZX",@"ZXY",@"XZY",@"YXZ",@"ZYX"])
#define DEFAULT_ORDER @"XYZ"

@implementation Euler

+ (instancetype)eulerWithX:(double)x y:(double)y z:(double)z order:(NSString *)order {
    Euler *euler = [[Euler alloc] initWithX:x y:y z:z order:order];
    return euler;
}

- (instancetype)initWithX:(double)x y:(double)y z:(double)z order:(NSString *)order{
    if (self = [self init]) {
        [self setValueWithX:x y:y z:z order:order];
    }
    return self;
}

- (void)setValueWithX:(double)x y:(double)y z:(double)z order:(NSString *)order {
    self.x = x;
    self.y = y;
    self.z = z;
    self.order = [ROTATION_ORDERS containsObject:order] ? order : DEFAULT_ORDER;
}

@end
